packages/components/date-picker/src/date-picker.js

```diff
@@ -0,0 +1,281 @@
+import { customElementDefineHelper, html, TSElement, unsafeCSS } from '@tradeshift/elements';
+import '@tradeshift/elements.date-picker-overlay';
+import '@tradeshift/elements.text-field';
+import '@tradeshift/elements.button';
+import '@tradeshift/elements.icon';
+// eslint-disable-next-line import/no-duplicates
+import '@tradeshift/elements.overlay';
+// eslint-disable-next-line import/no-duplicates
+import { TSOverlay } from '@tradeshift/elements.overlay';
+import css from './date-picker.css';
+import { translations } from './utils';
+
+export class TSDatePicker extends TSElement {
+	constructor() {
+		super();
+		this._translations = Object.assign({}, translations);
+		this.pageDate = new Date().toISOString();
+		this.opened = false;
+		this.typeable = false;
+		this.firstDay = 1;
+	}
+
+	static get styles() {
+		return [TSElement.styles, unsafeCSS(css)];
+	}
+
+	static get properties() {
+		return {
+			/** Can be used for customizing placeholder, days abbreviations, months abbreviations and providing translations for them
+			 * see the structure in its storybook knobs section.
+			 * */
+			translations: { type: Object, reflect: true },
+			/** Direction of the component 'rtl' or 'ltr'. */
+			dir: { type: String, reflect: true },
+			/** For setting the date of the date picker you can use this prop/attribute. It will get update after the user changes the date. */
+			selectedDate: { type: String, reflect: true, attribute: 'selected-date' },
+			/** This date can be used as a way to customize the month that date picker dropdown will be opened in,
+			 *  by default it's the current month. You can pass any date in the preferred month. */
+			pageDate: { type: String, reflect: true, attribute: 'page-date' },
+			/** INTERNAL */
+			selectedDateObj: { type: String, reflect: false },
+			/** Label of the date picker. */
+			label: { type: String, reflect: true },
+			/** Is it possible to deselect the date? You will be able to change the date still. */
+			deselectable: { type: Boolean, reflect: true },
+			/** Is the date picker is disabled? */
+			disabled: { type: Boolean, reflect: false },
+			/** Is the date picker is readonly? */
+			readonly: { type: Boolean, reflect: false },
```

Comment:
I not see any mockups for readonly state. As a matter of fact, in readonly state your input is not disabled and you still able to type in or open overlay and change the date, it just renders a lock icon on the right side of the input :)

Reply:
Besides that, what a description :)) `Is the date picker is readonly?` :))))
I'll fix it and add the test and knob to storybook. Nice catch.